src/cephadm/cephadm

```diff
@@ -2499,6 +2536,67 @@ def command_adopt():
     else:
         raise Error('adoption of style %s not implemented' % args.style)
 
+def command_adopt_prometheus(daemon_id, uid, gid):
+    # type: (str, int, int) -> None
+
+    daemon_type = 'prometheus'
+    if args.style != 'legacy':
+        raise Error('adoption of style %s not implemented' % args.style)
+
+    fsid = get_legacy_daemon_fsid(args.cluster,
+                                  daemon_type,
+                                  daemon_id,
+                                  legacy_dir=args.legacy_dir)
+    if not fsid:
+        raise Error('could not detect legacy fsid; set fsid in ceph.conf')
+
+    l = FileLock(fsid)
+    l.acquire()
+
+    unit_name = 'prometheus'
+    (enabled, state) = check_unit(unit_name)
+
+    if state == 'running':
+        logger.info('Stopping old systemd unit %s...' % unit_name)
+        call_throws(['systemctl', 'stop', unit_name])
+    if enabled:
+        logger.info('Disabling old systemd unit %s...' % unit_name)
+        call_throws(['systemctl', 'disable', unit_name])
+
+    data_dir_dst = make_data_dir(fsid, daemon_type, daemon_id,
+                                     uid=uid, gid=gid)
+
+    # config
+    config_src = '/etc/prometheus/prometheus.yml'
+    config_src = os.path.abspath(args.legacy_dir + config_src)
+    config_dst = os.path.join(data_dir_dst, 'etc/prometheus')
+    copy_files([config_src], config_dst, uid=uid, gid=gid)
+
+    # data
+    data_src = '/var/lib/prometheus/metrics/'
+    data_src = os.path.abspath(args.legacy_dir + data_src)
+    data_dst = os.path.join(data_dir_dst, 'data')
+    copy_tree([data_src], data_dst, uid=uid, gid=gid)
+
+    m = Monitoring.components[daemon_type]  # type: ignore
+    metadata = m.get('image', dict())  # type: ignore
+    monitoring_args = [
+        '--user',
+        str(uid),
+        # FIXME: disable cpu/memory limits for the time being (not supported
+        # by ubuntu 18.04 kernel!)
+        #'--cpus',
+        #m.get('cpus', '2'),  # type: ignore
+        #'--memory',
+        #m.get('memory', '4GB')  # type: ignore
```

Comment:
this is copy pasta from command_deploy().. can we avoid duplicating this?  maybe we can just push it down into get_container()?

Reply:
yeah, I saw that it got commented and didn't know whether to remove it outright or leave it for whatever the decision was going to be.

Reply:
It's not so much the commented part, but this whole special case for prometheus where we need extra args.  This list should only appear in one place.

Reply:
I agree... but I didn't want to get too carried away in the one PR.. I figured the uid_gid_monitoring might be pushing it. :)

If you want I can look at refactoring that part in this PR... or submit another to keep this one short.

Reply:
:+1: for moving this down into `get_container` instead of passing via `container_args`

it's similar to how the `ceph_args` are handled ... https://github.com/ceph/ceph/blob/6e05a926887dc368b70899b133048138b94de1fc/src/cephadm/cephadm#L1196

Reply:
done.